classes/dns_util.php

```diff
@@ -96,6 +96,26 @@ public function get_subdomains($domain) {
         return rtrim(strstr($domain, $primarydomain, true), '.');
     }
 
+    /**
+     * Gets the most common user email domains from the database.
+     * @param int $number limits the number of domains
+     * @return array
+     */
+    public function get_user_domains(int $number = 0): array {
+        global $DB;
+
+        $domainsql = $DB->sql_substr('LOWER(email)', $DB->sql_position("'@'", 'email') . ' + 1');
+        $sql = "SELECT $domainsql AS domain, count(*) AS count
+                  FROM {user}
+                 WHERE email LIKE '%@%.%' AND suspended = 0 AND deleted = 0
```

Comment:
just wondering about the LIKE and its performance if its needed. If we had a dodgy email in the system then in theory it never gets confirmed so maybe we can just filter on confirmed = 1 and that's all we need?

Reply:
In one of my test sites I have do have a handful of accounts that are confirmed without valid emails, one of them being the guest user. The others are possibly related to some manual change, but removing the LIKE opens it up the possibility of some weird results.

Reply:
ok cool